src/service.c

```diff
@@ -170,11 +171,16 @@ static status_e activate_rpc( struct service *sp )
    if ( SC_PROTOVAL ( scp ) == IPPROTO_TCP ) {
       M_SET ( scp->sc_xflags, SF_NOLIBWRAP );
    }
+   if ( !service_port ) {
+      service_port = 0 ;
```

Comment:
this logic doesn't make sense ... the only way this `if` statement was true is if `service_port` was already `0`

i think what you want to write:
`service_port = SC_SPECIFIED( scp, A_PORT ) ? SC_PORT( scp ) : 0;`